src/fn-encoding.md

```diff
@@ -0,0 +1,109 @@
+# Motivation
+
+We'd like to be able to use something similar to `unstable-fn`, but without deep support in core.
+We'd also like to experiment with know what kinds of programming can be encoded using egglog.
+
+# Constraints
+
+- Primitives can't have type information
+- Support for the REPL (not a whole-program transformation)
+
+# Known limitations
```

Comment:
Do you think this approach makes resugaring for proofs harder? Or is it about as hard as it was before?

Reply:
I'm starting to think that proofs / visualizations are no harder or easier whatever we decide